Source/Editor/Content/GUI/ContentView.cs

```diff
@@ -714,54 +714,68 @@ public override bool OnCharInput(char c)
-                int itemsToFit = Mathf.FloorToInt(width / defaultItemsWidth) - 1;
-                if (itemsToFit < 1)
-                    itemsToFit = 1;
-                int xSpace = 4;
-                float itemsWidth = width / Mathf.Max(itemsToFit, 1) - xSpace;
-                float itemsHeight = itemsWidth / defaultItemsWidth * (ContentItem.DefaultHeight * viewScale);
-                var flooredItemsWidth = Mathf.Floor(itemsWidth);
-                var flooredItemsHeight = Mathf.Floor(itemsHeight);
-                x = itemsToFit == 1 ? 1 : itemsWidth / itemsToFit + xSpace;
-                for (int i = 0; i < _children.Count; i++)
-                {
-                    var c = _children[i];
-                    c.Bounds = new Rectangle(Mathf.Floor(x), Mathf.Floor(y), flooredItemsWidth, flooredItemsHeight);
+                case ContentViewType.Tiles:
+                    float xSpace = ContentItem.DefaultMarginSize * viewScale;
+                    float ySpace = ContentItem.DefaultMarginSize * viewScale;
+                    float itemsWidth = ContentItem.DefaultWidth * viewScale ;
+                    itemsHeight = ContentItem.DefaultHeight * viewScale;
+                    //width - element / element gives elementRowCount, * it by element width gives actual size
+                    //where last elemet is - it by from width gives offset to the edge now because 
+                    //the for loop contans x += itemsWidth + xSpace;
+                    //so if all of it gets / by element count it will give the scaling
+                    float elementRowCount = Mathf.Floor(width / (itemsWidth + xSpace));
+                    float lastElement = Mathf.Max((elementRowCount * (itemsWidth + xSpace)), 0);
+                    if (_children.Count > elementRowCount)
+                    {
+                        
+                        float lastElementDistanceFromEdge = width - lastElement;
+                        float scalePerElement = (lastElementDistanceFromEdge / (elementRowCount + 1));
+                        itemsWidth += scalePerElement;
+                        itemsHeight += scalePerElement;
+                        lastElement -= itemsWidth + xSpace;
+                    }
 
-                    x += (itemsWidth + xSpace) + (itemsWidth + xSpace) / itemsToFit;
-                    if (x + itemsWidth > width)
+                    var flooredItemsWidth = Mathf.Floor(itemsWidth);
+                    var flooredItemsHeight = Mathf.Floor(itemsHeight);
+                    x = ContentItem.DefaultMarginSize;
+                    for (int i = 0; i < _children.Count; i++)
                     {
-                        x = itemsToFit == 1 ? 1 : itemsWidth / itemsToFit + xSpace;
-                        y += itemsHeight + 7;
+                        var c = _children[i];
+                        c.Bounds = new Rectangle(x,y, flooredItemsWidth, flooredItemsHeight);
```

Comment:
It may be worth double checking to make sure that x and y are always ints. May have to Floor them here so we dont get half pixels. Ran into this on the first round (granted we were dividing) where they were now always whole numbers.

Reply:
the Bounds is rect with has float2 as x and y ruining the Floor for every element can add up and having stuff shifted by 1 pixel it will happen for all elements it is not something critical